### ISSUE TITLE
Multiple Tokens for requests with .retry()

### ISSUE THREAD

**Issue by MarkusFalgenhauer** [benbaran/adal-angular4#108]
Hi,

I am still on angular4 Version 1.1.11 but i think this will also affect newer Versions.
When i do a .retry() on the http request, with every retry a Bearer token is added to the Authorization Header.
So: first call is correct, second has: Authorization: Bearer ey......., Bearer ey.... in the header and so on, which leads to Unauthorized.

Is there a quick solution to avoid this behavior?

**Comment by benbaran** [comment]
Yeah, this is definitely a bug. The intercepter needs to be modified to replace the header instead of appending to it. I’m out of town, but will take a look when I get back.

Or if you or anyone else wants to take a crack at it, I’ll gladly accept the contribution: )

**Comment by geerzo** [comment]
I think this issue is fixed in the latest code as it uses "headers.set" which should replace, not append. There was a time earlier this year where it was append though.

**Comment by benbaran** [comment]
I was able to reproduce. Maybe “set” doesn’t work as expected?

**Comment by geerzo** [comment]
That is possible, or it's a complication with using clone at the same time? Docs seem to state it should work. Wonder if there is a bug in certain angular versions?

https://angular.io/api/http/Headers#set

I remember having this problem before but once it was changed from append to set it started working as designed (again, latest code, not 1.1.11). @benbaran do you have sample code you used to reproduce?

**Comment by benbaran** [comment]
Looks like I was looking at an older version. I'm thinking you are right this may be fixed.
